frame/society/src/lib.rs

```diff
@@ -1034,47 +962,64 @@ decl_module! {
 		///
 		/// Total Complexity: O(1)
 		/// # </weight>
-		#[weight = T::BlockWeights::get().max_block / 10]
-		fn set_max_members(origin, max: u32) {
+		#[pallet::weight(T::BlockWeights::get().max_block / 10)]
+		pub fn set_max_members(origin: OriginFor<T>, max: u32) -> DispatchResult {
 			ensure_root(origin)?;
 			ensure!(max > 1, Error::<T, I>::MaxMembers);
-			MaxMembers::<I>::put(max);
-			Self::deposit_event(RawEvent::NewMaxMembers(max));
+			<MaxMembers<T, I>>::put(max);
+			Self::deposit_event(Event::NewMaxMembers(max));
```

Comment:
need to come back to this, but thinking this is probably not fully right, and that there should also be some upper limit to the max members